### PR TITLE
added everything

### PR DESCRIPTION
**Pull request by FennaHD**
This change is [Reviewable](https://reviewable.io/reviews/fennahd/pokedex/1)

### CODE REVIEW

File: app/src/androidTest/java/com/example/pokedex/ExampleInstrumentedTest.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.junit.Assert.*
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
```
Comment (didi9) [inline app/src/androidTest/java/com/example/pokedex/ExampleInstrumentedTest.kt:17]: Did you throw this in?

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:supportsRtl="true"
+        android:theme="@style/AppTheme">
+        <activity android:name=".MainActivity">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
```
Comment (didi9) [inline app/src/main/AndroidManifest.xml:15]: Remove newline

File: app/src/main/java/com/example/pokedex/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import android.view.Gravity
+import android.widget.Button
+import android.widget.LinearLayout
+import kotlinx.android.synthetic.main.activity_main.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class MainActivity : AppCompatActivity() {
+
```
Comment (didi9) [inline app/src/main/java/com/example/pokedex/MainActivity.kt]: remove newline

**Review by didi9 — CHANGES_REQUESTED** [review]
> Reviewed 123 of 124 files at r1.
> *[Reviewable](https://reviewable.io/reviews/fennahd/pokedex/1)* status: 123 of 124 files reviewed, 22 unresolved discussions (waiting on @didi9 and @FennaHD)

___
*[app/build.gradle, line 12 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh6exKBoC-Ql2-Qu8R:-Lzh6exKBoC-Ql2-Qu8S:b-z12oic) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/build.gradle#L12)):*
> ```Gradle
>     defaultConfig {
>         applicationId "com.example.pokedex"
>         minSdkVersion 15
> ```

Our minSdkVersion is 21
___
*[app/src/androidTest/java/com/example/pokedex/ExampleInstrumentedTest.kt, line 17 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh6npNAcTwWFgSkQeA:-Lzh6npNAcTwWFgSkQeB:btn5gjd) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/androidTest/java/com/example/pokedex/ExampleInstrumentedTest.kt#L17)):*
> ```Kotlin
>  */
> @RunWith(AndroidJUnit4::class)
> class ExampleInstrumentedTest {
> ```

Did you throw this in?
___
*[app/src/main/AndroidManifest.xml, line 15 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh6soo3xhebschSc6E:-Lzh6soo3xhebschSc6F:b-fgj26b) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/AndroidManifest.xml#L15)):*
> ```XML
>             <intent-filter>
>                 <action android:name="android.intent.action.MAIN" />
> ```

remove newline
___
*[app/src/main/java/com/example/pokedex/IndividualActivity.kt, line 17 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhOT-M0Yb5uWqPrfLS:-LzhOT-NEmWShINPqaDB:b-ry29lv) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/IndividualActivity.kt#L17)):*
> ```Kotlin
>         super.onCreate(savedInstanceState)
>         setContentView(R.layout.individual_poke_view)
>         val name = intent.getStringExtra("name")
> ```

put into companion object
___
*[app/src/main/java/com/example/pokedex/IndividualActivity.kt, line 21 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhPJyi2mb4dgSbWWUv:-LzhPJyi2mb4dgSbWWUw:b-vzxus7) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/IndividualActivity.kt#L21)):*
> ```Kotlin
>         //action bar for loading previous activity
>         val actionbar = supportActionBar
>         actionbar!!.title = "Back"
> ```

use null check (`actionbar?.title`) instead of double-bang
___
*[app/src/main/java/com/example/pokedex/IndividualActivity.kt, line 21 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhPRRR2ncwH_dot04k:-LzhPRRR2ncwH_dot04l:b-e2o92) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/IndividualActivity.kt#L21)):*
> ```Kotlin
>         //action bar for loading previous activity
>         val actionbar = supportActionBar
>         actionbar!!.title = "Back"
> ```

Try naming it the name of the Pokemon
___
*[app/src/main/java/com/example/pokedex/IndividualActivity.kt, line 23 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhQ_HlECASRPmt6wBw:-LzhQ_HlECASRPmt6wBx:b-5elkdt) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/IndividualActivity.kt#L23)):*
> ```Kotlin
>         actionbar!!.title = "Back"
>         actionbar.setDisplayHomeAsUpEnabled(true)
>         actionbar.setDisplayHomeAsUpEnabled(true)
> ```

remove one
___
*[app/src/main/java/com/example/pokedex/IndividualActivity.kt, line 34 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhR6TPE98KI7tZQhqq:-LzhR6TQ1hwuo9s8uXLC:bg778d) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/IndividualActivity.kt#L34)):*
> ```Kotlin
>                 val imageView = findViewById<ImageView>(R.id.pokemonPicture)
>                 Picasso.get().load(response.body()?.sprites?.front_default).resize(400, 400).into(imageView)
>                 findViewById<TextView>(R.id.name).text = getString(R.string.pokemon_name, response.body()?.name?.capitalize())
> ```

use kotlin synthetics (e.g. `name.text = ...`)
___
*[app/src/main/java/com/example/pokedex/IndividualActivity.kt, line 37 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhRgrE01BZKXTcyR9U:-LzhRgrE01BZKXTcyR9V:bgo2i51) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/IndividualActivity.kt#L37)):*
> ```Kotlin
>                 findViewById<TextView>(R.id.weight).text = getString(R.string.pokemon_weight, response.body()?.weight)
>                 findViewById<TextView>(R.id.height).text = getString(R.string.pokemon_height, response.body()?.height)
>                 val sb = StringBuilder("Type: ")
> ```

try using `.joinToString()`
___
*[app/src/main/java/com/example/pokedex/MainActivity.kt, line 16 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhHl-6EBOzPzW9O6ac:-LzhHl-7E0IuGbAmGjeU:b-y0zyru) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/MainActivity.kt#L16)):*
> ```Kotlin
> import retrofit2.Response
> 
> class MainActivity : AppCompatActivity() {
> ```

reformat (let's compare formatting settings)
___
*[app/src/main/java/com/example/pokedex/MainActivity.kt, line 17 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh6wPKApd0KNGhOS0Z:-Lzh6wPKApd0KNGhOS0_:b-fgj26b) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/MainActivity.kt#L17)):*
> ```Kotlin
> 
> class MainActivity : AppCompatActivity() {
> ```

remove newline
___
*[app/src/main/java/com/example/pokedex/MainActivity.kt, line 24 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh7JzC7CDYJ6wMJVqd:-Lzh7JzC7CDYJ6wMJVqe:b-i33n9l) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/MainActivity.kt#L24)):*
> ```Kotlin
> 
>         var pokemonList: List<PokemonSummary>?
>         val pokedexService = PokeApiService.create()
> ```

You could inline both of these
___
*[app/src/main/java/com/example/pokedex/MainActivity.kt, line 31 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhBPMH6uWlCOOXhes7:-LzhBPMH6uWlCOOXhes8:bkkd11i) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/MainActivity.kt#L31)):*
> ```Kotlin
>                 pokemonList = response.body()?.results
>                 pokemonList?.forEach {
>                     constraintLayout.addView(createButton(this@MainActivity, it.name))
> ```

Interesting solution! This is an instance where we would use a RecyclerView. Check out RecyclerViews and see if you can change all of this to use a RecyclerView. Let me know if you have any questions
___
*[app/src/main/java/com/example/pokedex/MainActivity.kt, line 41 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh7dZG-pHTe33ujRVe:-Lzh7dZG-pHTe33ujRVf:bpxhrp4) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/MainActivity.kt#L41)):*
> ```Kotlin
> 
>     fun createButton(context: Context, name: String): Button {
>         var button = Button(context)
> ```

use `.apply { ... }`
___
*[app/src/main/java/com/example/pokedex/MainActivity.kt, line 51 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh7juaAjsZH2lMarm4:-Lzh7juaAjsZH2lMarm5:b-gpijph) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/MainActivity.kt#L51)):*
> ```Kotlin
>             val intent = Intent(context, IndividualActivity::class.java)
>             intent.putExtra("name", name)
>             startActivity(intent)
> ```

Inline the intent and you can put the `putExtra` on when you create it (e.g. `Intent(..., ...).putExtra(...)`)
___
*[app/src/main/java/com/example/pokedex/PokeApiService.kt, line 11 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh8L7W-GXeS--bCZDh:-Lzh8L7W-GXeS--bCZDi:b-z58w7t) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/PokeApiService.kt#L11)):*
> ```Kotlin
> interface  PokeApiService{
>     @GET("pokemon/?offset=0&limit=964")
>     fun retrieveData() : Call<PokedexData>
> ```

reformat (`retrieveData(): Call<PokedexData>`). Same with the next call. We'll need to compare our format files so it'll do it automatically
___
*[app/src/main/java/com/example/pokedex/PokeApiService.kt, line 20 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh8adM-R6J0aKTCV0A:-Lzh8adM-R6J0aKTCV0B:b-o1gaak) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/PokeApiService.kt#L20)):*
> ```Kotlin
>             val retrofit = Retrofit.Builder()
>                     .baseUrl("https://pokeapi.co/api/v2/")
>                     .addConverterFactory(MoshiConverterFactory.create())
> ```

Change to use Gson (this library is fine, but just to match what we do now)
___
*[app/src/main/java/com/example/pokedex/PokeApiService.kt, line 23 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhEHW32jk6wQOcK-fP:-LzhEHW48EzBsdYRyIXz:b-qrszgs) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/PokeApiService.kt#L23)):*
> ```Kotlin
>                     .build()
> 
>             return retrofit.create(PokeApiService::class.java)
> ```

You can add the `.create()` onto the end (like on the line after `.build()`) and then return the whole thing instead of the variable
___
*[app/src/main/java/com/example/pokedex/Response.kt, line 3 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh9LFU5SxTovPXZNL6:-Lzh9LFU5SxTovPXZNL7:b8qx0fo) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/Response.kt#L3)):*
> ```Kotlin
> package com.example.pokedex
> 
> data class PokedexData(
> ```

data classes look good! we normally put each data class into their own `.kt` file unless it's just the wrapper
___
*[app/src/main/java/com/example/pokedex/Response.kt, line 21 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-Lzh928v-STzcdPCPX-g:-Lzh928v-STzcdPCPX-h:b-6j731y) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/java/com/example/pokedex/Response.kt#L21)):*
> ```Kotlin
> )
> 
> data class TypeArray(
> ```

Probably just name this `Type`, since when you make the list we'll see that it's a list of Types (not a list of TypeArrays)
___
*[app/src/main/res/layout/activity_main.xml, line 2 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhF0c861FQ8AoYGNuT:-LzhF0c861FQ8AoYGNuU:b7ak3b) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/res/layout/activity_main.xml#L2)):*
> ```XML
> <?xml version="1.0" encoding="utf-8"?>
> <ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
> ```

We'll move this to a ConstraintLayout  with a RecyclerView in it
___
*[app/src/main/res/layout/individual_poke_view.xml, line 2 at r1](https://reviewable.io/reviews/fennahd/pokedex/1#-LzhFJSxDFXiQdnQvbD7:-LzhFJSxDFXiQdnQvbD8:bq7lxc1) ([raw file](https://github.com/fennahd/pokedex/blob/aa432cf1e99aca37761fe414092493f02911e804/app/src/main/res/layout/individual_poke_view.xml#L2)):*
> ```XML
> <?xml version="1.0" encoding="utf-8"?>
> <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
> ```

Change to ConstraintLayout

**Review by didi9 — COMMENTED** [review]
> Reviewed 1 of 124 files at r1.
> *[Reviewable](https://reviewable.io/reviews/fennahd/pokedex/1)* status: all files reviewed, 22 unresolved discussions (waiting on @FennaHD)

**Review by romrell4 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/fennahd/pokedex/1)* status: 101 of 125 files reviewed, 26 unresolved discussions (waiting on @didi9, @FennaHD, and @romrell4)

___
*[.idea/.name, line 1 at r2](https://reviewable.io/reviews/fennahd/pokedex/1#-LzwHDfK4etFITM7HdIc:-LzwHDfK4etFITM7HdId:b-lsv9g7) ([raw file](https://github.com/fennahd/pokedex/blob/f53dbb3c9d577fa8fa6f95e319104f5b7e0e066d/.idea/.name#L1)):*
> ```
> Pokedex
> ```

We gitignore the .idea file. Do you know how to use gitignore?

**Review by romrell4 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/fennahd/pokedex/1)* status: 101 of 125 files reviewed, 27 unresolved discussions (waiting on @didi9, @FennaHD, and @romrell4)

___
*[Pokedex.iml, line 1 at r2](https://reviewable.io/reviews/fennahd/pokedex/1#-LzwHJnL6sUqEo_mSjoo:-LzwHJnL6sUqEo_mSjop:bc4dgxv) ([raw file](https://github.com/fennahd/pokedex/blob/f53dbb3c9d577fa8fa6f95e319104f5b7e0e066d/Pokedex.iml#L1)):*
> ```XML
> <?xml version="1.0" encoding="UTF-8"?>
> ```

We gitignore the *.iml files

**Pull request closed without merging**